### ISSUE TITLE
(Class/field) should warn

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#2260]
The `(Class/field)` interop form works, but only accidentally and it's undocumented. Because of changes in clojure 1.12 it will no longer work and kondo (or lsp) should warn about it.

cc @fogus @puredanger @ericdallo

**Comment by borkdude** [comment]
Note to self:

Currently clj-kondo can retrieve info about `.java` or `.class` files
Example:

```
(def x (source-is->java-member-definitions (io/input-stream "/Users/borkdude/.cache/clojure-lsp/jdk/java.base/java/lang/System.java") "/Users/borkdude/.cache/clojure-lsp/jdk/java.base/java/lang/System.java"))
```

<img width="1499" alt="Screenshot 2024-01-19 at 21 07 09" src="https://github.com/clj-kondo/clj-kondo/assets/284934/85cf119e-f2db-4a1a-89b0-25407d2a79b1">

clj-kondo should store the stuff it needs for linting in `.clj-kondo/.cache/v1/java/java.lang.System.transit.json` or so.

Whenever a class is used we will retrieve info from that place and use it for linting.

**Comment by borkdude** [comment]
TODO:

- [ ] only write to cache if class definition already in idacs
- [ ] only read from cache if class definition isn't already in idacs
- [ ] the :interop? stuff returned from resolve-name can probably be just a nil value as not do unnessary work?
- [ ]
